pkg/api/types.go

```diff
@@ -2703,6 +2703,21 @@ type ServiceSpec struct {
 	// and ExternalTrafficPolicy is set to Local.
 	// +optional
 	HealthCheckNodePort int32
+
+	// TolerateUnreadyEndpoints, when set to true, indicates that endpoints
+	// controller should create endpoints for the unready Pods associated with
+	// the Service. This is useful for allowing the Pods associated with a
+	// headless Service to be DNS resolvable without respect to their readiness checks.
+	// Endpoints of these Services retain their DNS records and continue
+	// receiving traffic for the Service from the moment the kubelet starts all
+	// containers in the Pod and marks it "Running", until the kubelet stops all
```

Comment:
Ideally, we want the service to be resolvable also for pod phase "pending" + condition "initializing". Basically, we want the service to be resolvable in the init container.